packages/google-pay-integration/src/gateways/google-pay-gateway.ts

```diff
@@ -31,6 +31,7 @@ import itemsRequireShipping from '../utils/items-require-shipping';
 
 export default class GooglePayGateway {
     private _getPaymentMethodFn?: () => PaymentMethod<GooglePayInitializationData>;
+    private _isBuyNowFlow = false;
```

Comment:
Just want to clarify. Why do we need this `_isBuyNowFlow` variable?

Reply:
The Buy Now flow is only related to the PDP and we need to detect it. We need to correctly collect `paymentDataRequest` for GooglePay when we are on a product detail page that does not contain checkout and cart information. This variable helps us understand that we are on the product details page.

Reply:
Do we have only Braintree Google Pay button on the PDP?

Reply:
I'm not sure for 100%, but I think would be better to have separate gateway/strategy/processor for the PDP.
@serhii-tkachenko you worked a lot more with these buttons, what do you think?

Reply:
Yes, currently there is only Braintree Google Pay button on the PDP, other GP buttons on this page are not available.

Reply:
This is a known behaviour for making one button strategy to cover both primary and secondary (buy now) carts. We have a point in checkout improvements list to find a way how to build buttons strategies to work with Buy Now flow.

cc @animesh1987